src/tools/CMakeLists.txt

```diff
@@ -0,0 +1,261 @@
+set(HUNSPELL_TOOLS_HZIP_TARGET      "${HUNSPELL_TOOLS_TARGET_PREFIX}_${_HZIP_OUTPUT_NAME}")
+set(HUNSPELL_TOOLS_MUNCH_TARGET     "${HUNSPELL_TOOLS_TARGET_PREFIX}_${_MUNCH_OUTPUT_NAME}")
+set(HUNSPELL_TOOLS_UNMUNCH_TARGET   "${HUNSPELL_TOOLS_TARGET_PREFIX}_${_UNMUNCH_OUTPUT_NAME}")
+
+set(HUNSPELL_TOOLS_OUTPUT_PREFIX    ""  CACHE STRING "Prefix added to binaries output filename.")
+mark_as_advanced(HUNSPELL_TOOLS_OUTPUT_PREFIX)
+# ---------------------------------------------------------------------------- #
+
+
+# ---------------------------------------------------------------------------- #
+# ~~~ Hunspell target
+
+# ~~ Add executable
+add_executable(${HUNSPELL_TOOLS_HUNSPELL_TARGET} hunspell.cxx)
+
+# ~~~ Check for headers
+include(CheckIncludeFile)
+check_include_file(locale.h     _HAVE_LOCALE_H)
+check_include_file(langinfo.h   _HAVE_LANGINFO_H)
+
+# ~~~ Find libraries  
+
+# ~ Iconv
+if(UNIX)
+    find_package(Iconv REQUIRED)
+endif()
+
+# ~ Curses
+if(UNIX)
+    option(HUNSPELL_TOOLS_USE_CURSES  "Locate and link curses library."  OFF)
+
+    if(HUNSPELL_TOOLS_USE_CURSES)
+        set(CURSES_NEED_WIDE ON)
+        find_package(Curses)
+
+        if(NOT CURSES_FOUND)
+            message(FATAL_ERROR  "Could not find curses library. Set paths manually.")
+        endif()
+    endif()
+endif()
+
+
+if(CURSES_FOUND)
+    set(_HAVE_CURSES_H    OFF)
+    set(_HAVE_NCURSESW_H  OFF)
+    set(_CURSES_INC_DIR   "")
+    set(_CURSES_LIB       "")
+
+    get_filename_component(_CURSES_LIB_NAME  ${CURSES_NCURSES_LIBRARY}  NAME)
+
+    # libcurses just points to libncurses in most cases
+    if(EXISTS "${CURSES_INCLUDE_PATH}/curses.h")
+        set(_HAVE_CURSES_H   ON)
+        set(_CURSES_INC_DIR  "${CURSES_INCLUDE_PATH}")
+        set(_CURSES_LIB      "${CURSES_CURSES_LIBRARY}")
+    endif()
+    if(EXISTS "${CURSES_INCLUDE_PATH}/ncurses/curses.h")
+        set(_HAVE_CURSES_H   ON)
+        set(_CURSES_INC_DIR  "${CURSES_INCLUDE_PATH}/ncurses")
+        set(_CURSES_LIB      "${CURSES_NCURSES_LIBRARY}")
+    endif()
+    if(EXISTS "${CURSES_INCLUDE_PATH}/ncursesw/curses.h" AND ${_CURSES_LIB_NAME} MATCHES "w")
+        set(_HAVE_NCURSESW_H  ON)
+        set(_CURSES_INC_DIR   "${CURSES_INCLUDE_PATH}")
+        set(_CURSES_LIB       "${CURSES_NCURSES_LIBRARY}")
+    endif()
+endif()
+
+# ~ Readline
+if(UNIX)
+    option(HUNSPELL_TOOLS_USE_READLINE  "Locate and link readline library."  OFF)
+endif()
+
+if(HUNSPELL_TOOLS_USE_READLINE)
+    find_path(READLINE_INCLUDE_PATH
+        NAMES  readline/readline.h
+        DOC    "Readline include directory."
+    )
+    find_library(READLINE_LIBRARY
+        NAMES  readline
+        DOC    "Readline library."
+    )
+    mark_as_advanced(
+        READLINE_INCLUDE_PATH
+        READLINE_LIBRARY
+    )
+    find_package_handle_standard_args(Readline
+        REQUIRED_VARS  
+            READLINE_LIBRARY
+            READLINE_INCLUDE_PATH
+    )
```

Comment:
What is this doing smack dab in the middle of everything? This is a function made available by [FindPackageHandleStandardArgs](https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html) for ONLY inside find modules. For one, the module is not included anywhere, but more importantly this code is not even in a find module.